pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -199,6 +203,48 @@ func E2ECommonSpec(ctx context.Context, inputGetter func() E2ECommonSpecInput) {
 				By(fmt.Sprintf("Get management cluster resources created by addons-manager for workload cluster %q on management cluster %q", clusterName, input.E2EConfig.ManagementClusterName))
 				clusterResources, err = getManagementClusterResources(ctx, mngclient, mngDynamicClient, mngAggregatedAPIResourcesClient, mngDiscoveryClient, namespace, clusterName, infrastructureName)
 				Expect(err).NotTo(HaveOccurred())
+
+				By(fmt.Sprintf("Upgrade workload cluster %q", clusterName))
```

Comment:
As commented above, `E2ECommonSpec` is not consumed by current TKGS E2E tests suite. All TKGS tests should be moved into https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/tkg/test/tkgctl/tkgs/